app/services/listing_index_service/search/zappy_adapter.rb

```diff
@@ -76,15 +85,29 @@ def listings_from_ids(id_obs, includes)
       Listing
         .where(id: ids)
         .order("field(listings.id, #{ids.join ','})")
-        .map {
-          |l| ListingIndexService::Search::Converters.listing_hash(l, includes)
+        .map { |l|
+          meta = Maybe(id_obs.find {|r| r['id'] == l.id.to_s })
```

Comment:
This should probably be done outside the `map`, because this seems O(n<sup>2</sup>) to me.

Maybe in the same pass as getting the `id`s?